docs/document-qa.md

```diff
@@ -0,0 +1,116 @@
+# Document Question and Answering
+
+## Overview
+
+One of the most popular use cases for LLMs is doing Q/A over a large corpus of unstructured data.
+
+Fixie makes it possible to quickly build Q/A Agents by automatically crawling a set of developer-provided URLs, generating embeddings, chunking the data, and storing it inside of a Vector Database for efficient retrieval.
+
+To understand how to get started, let's look at a quick example of a [CodeShotAgent](/agents.md/#codeshotagent) that is designed to answer questions about Fixie:
+
+```python
+import fixieai
+
+BASE_PROMPT = """I am a helpful assistant that can answer questions \
+  about building on top of Fixie.ai. I try to be as concise as possible \
+  in my answers while still effectively answering the question."""
+
+URLS = [
+    "https://docs.fixie.ai/*"
+]
+
+DOCUMENTS = [fixieai.DocumentCorpus(urls=URLS)]
+agent = fixieai.CodeShotAgent(BASE_PROMPT, [], DOCUMENTS, conversational=True)
+```
+
+You'll notice that we created a new variable, `URLS`, that contains an array of webpages that we would like Fixie to automatically crawl. Notice the `*` at the end of the Fixie docs URL. This tells Fixie that we want to not just crawl the single page, but we want to crawl **all** webpages found on that base URL (similar to how a search engine works).
+
+Fixie also supports indexing individual pages, but the `*` can be useful when you have hundreds of pages that need to be ingested. If you need to exclude certain files (or types of files), see [Excluding documents during crawl](#excluding-documents-during-crawl).
+
+> Note: By default, Fixie will crawl up to 1,000 documents for each wildcard URL. If you need more, [get in touch](mailto:hello@fixie.ai).
+
+Once we have our `URLS`, we instantiate a new `CodeShotAgent` with our `BASE_PROMPT`, an empty Few Shot array, a set of `DOCUMENTS` that point to our URLs, and whether we want a `conversational` Agent (conversational agents keep previus questions and answers in memory, so we recommend it for Q/A Agents).
+
+When a `DocumentCorpus` is provided, FewShots are optional (hence the empty array that we passed in). If you need FewShots, see [Using Fewshots with Docs](#using-fewshots-with-docs).
+
+Once ready, you can deploy your agent using `fixie deploy`. Note that depending on the size and number of documents, indexing can take some time. While indexing is in progress, your agent will deploy but will always return, `I'm still starting up. Please try again in a few minutes.` See [Monitoring Indexing Status](#monitoring-indexing-status) for more.
```

Comment:
> can take some time.

Can we be a little more specific / give an order of magnitude? Minutes / hours / days?

Reply:
Unfortunately, I don't think so. It really depends.

Reply:
Ok, then maybe we could say something like "seconds for 10s of pages; minutes for 100s of pages; hours for 1000s of pages"?

Reply:
Giving misleading answers feels worse than giving no answer. Once we have more data, I think we can circle back. For now, I'll add a sentence that says, "This could take upwards of a few hours for large data sets"